components/woo/order.php

```diff
@@ -1633,32 +1633,45 @@ private function get_package_carrier_map() {
 			'parcel_letter' => array( 'dhl', 'dpd' ),
 		);
 	}
-	
+
+	/**
+	 * @param array $data
+	 *
+	 * @return mixed
+	 */
 	private function handle_email_notification( $data ) {
-		$carrier_email = $this->get_carrier_mail();
+		$carrier_email      = $this->get_carrier_mail();
 		$notification_email = $this->get_notification_email();
-		
-		if ( $data['carrier'] == 'dpd' || $data['carrier'] == 'dhl' ) {
-			if ( ! empty ( $carrier_email ) ) {
-				unset($data['notification_email']);
 
+		if ( ! empty ( $notification_email ) ) {
+			// Set fallback for the notification mail
+			$data['notification_email'] = $this->get_notification_email();
+		}
+
+		if ( ( ! isset( $data['carrier'] ) || $data['carrier'] !== 'dpd' )
+			 && ( ! isset( $data['carrier'] ) || $data['carrier'] !== 'dhl' )
+		) {
```

Comment:
^^ Isn't this a bit too much? If `$data['carrier']` isn't set, why should we check if it has a specific value?

Shouldn't it then be more like
```php
if ( 
  ! isset( $data['carrier'] ) || 
  ( $data['carrier'] !== 'dpd' || $data['carrier'] !== 'dhl' )
)
```